app/grandchallenge/retina_api/urls.py

```diff
@@ -77,4 +82,18 @@
         name="image-element-spacing-view",
     ),
     path("annotation/<int:user_id>/", include(annotation_router.urls)),
+    path(
+        "image/thumbnail/<uuid:image_id>/",
+        cache_page(settings.RETINA_IMAGE_CACHE_TIME)(
+            views.B64ThumbnailAPIView.as_view()
+        ),
+        name="image-thumbnail",
+    ),
+    path(
+        "image/thumbnail/<uuid:image_id>/<int:width>/<int:height>/",
```

Comment:
Any chance we can cause an OOM by setting large values here?

Reply:
Interesting, I had not thought about that. I just checked the [Pillow source](https://github.com/python-pillow/Pillow/blob/acf1c8559b720ec29f2438d7fb652e0dcf6cb0fb/PIL/Image.py#L1779) and apparently, it will not enlarge the image when the dimensions that are passed are larger than those of the image itself. So I think we are safe.